specs/sequencer-app.md

```diff
@@ -41,143 +42,309 @@ idea of "messages", where every transaction can contain multiple messages. The
+  | sudo_address | `Address` | The address which has authority over the bridge|
+  ||| account. If empty, assigned to the signer. |
+  | withdrawer_address | `Address` | The address which is allowed to withdraw funds|
+  ||| from the bridge account. If empty, assigned to the signer.|
+
+* `BridgeLock`: transfers funds from a sequencer account to a bridge account.
+It is effectively similar to `Transfer`. Upon execution of a bridge lock action,
+a `Deposit` event will be included in the block data for the rollup this bridge
+account is registered to, containing the information of the transfer.
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | to        | `Address`| The bridge account to transfer to. The "from" address|
+  ||| is assumed to be the signer of the transaction. |
+  | amount    | `uint128`| The amount to transfer. |
+  | asset     | `string` | The asset to transfer. |
+  | fee_asset | `string` | The asset used to pay for the action's fees. |
+  | destination_chain_address | `string` | The address on the destination chain|
+  ||| which will receive the bridged funds. |
+
+* `BridgeUnlock`: transfers funds from a bridge account to a sequencer account.
+Effectively similar to `Transfer`, it contains the following fields:
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | to        | `Address`| The account to transfer to. |
+  | amount    | `uint128`| The amount to transfer. |
+  | fee_asset | `string` | The asset used to pay for the action's fees. |
+  | memo      | `string` | Can be used to provide unique, identifying information|
+  ||| about the bridge unlock action. |
+  | bridge_address | `Address` | The address of the bridge account to transfer|
+  ||| from. |
+  | rollup_block_number | `uint64` | The block number on the rollup which triggered|
+  ||| the transaction underlying the bridge unlock. |
+  | rollup_withdrawal_event_id | `string` | An identifier of the original rollup|
+  ||| can be used to trace distinct rollup events from the bridge. |
+
+* `BridgeSudoChange`: changes the sudo and/or withdrawer address for the given
+bridge account. The signer must be the current bridge sudo account.
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | bridge_address | `Address` | The address of the bridge account to which these|
+  ||| changes should be made. |
+  | new_sudo_address | `Address` | The new sudo address for the bridge account.|
+  ||| If unset, will stay the same. |
+  | new_withdrawer_address | `Address` | The new withdrawer address for the bridge|
+  ||| account. If unset, will stay the same. |
+  | fee_asset | `string` | The asset with which to pay fees for this action.|
+
+### IBC User Actions
+
+Actions which deal with transfering funds between the sequencer and a separate
+chain.
+
+* `IbcRelay`: transmits data packets between the sequencer chain and another
```

Comment:
note that right now this is permissioned, only authorized accounts can relay, and this action is free to submit because of that. this also permissions who is allowed to open ibc channels